build-tools/packages/build-cli/src/commands/update-dep.ts

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { updatePackageJsonFile } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import execa from "execa";
+
+import { BaseCommand } from "../base";
+import { releaseGroupFlag } from "../flags";
+
+/**
+ * Updates the a transitive dependency in the lockfile.
+ *
+ * @remarks
+ * This command is primarily used manually to update transitive dependencies when we need to address CVEs.
+ *
+ * Note that this applies to all packages in the release group.
+ */
+export default class UpdateDepCommand extends BaseCommand<typeof UpdateDepCommand> {
+	static readonly description = `Updates a dependency in the lockfile to the latest version of a specified semver range.`;
+	static readonly enableJsonFlag = true;
+
+	static readonly flags = {
+		releaseGroup: releaseGroupFlag({ required: true }),
+		dependencyName: Flags.string({
+			description: "Name of the dependency (npm package) to update",
+			required: true,
+		}),
+		version: Flags.string({
+			description: "Semver range specifier to use when updating the dependency.",
+			required: true,
+		}),
+	};
+
+	public async run(): Promise<void> {
+		const context = await this.getContext();
+		const releaseGroup = context.repo.releaseGroups.get(this.flags.releaseGroup);
+
+		if (releaseGroup === undefined) {
+			// exits the process
+			this.error(`Can't find release group: ${this.flags.releaseGroup}`, { exit: 1 });
+		}
+
+		// Add override to package.json
+		this.info(`Adding pnpm override for ${this.flags.dependencyName}: ${this.flags.version} to package.json`);
+		updatePackageJsonFile(releaseGroup.directory, (json) => {
+			if (json.pnpm === undefined) {
+				json.pnpm = {};
+			}
+			/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+			if (json.pnpm.overrides === undefined) {
+				json.pnpm.overrides = {};
+			}
+			const currentOverrideVersion = json.pnpm.overrides[this.flags.dependencyName] as
+				| string
+				| undefined;
+			if (currentOverrideVersion !== undefined) {
+				this.error(
+					`A pnpm override for the specified dependency already exists (${this.flags.dependencyName}: "${currentOverrideVersion}". Cannot continue.`,
+					{ exit: 1 },
+				);
+			}
+			json.pnpm.overrides[this.flags.dependencyName] = this.flags.version;
+			/* eslint-enable @typescript-eslint/no-unsafe-member-access */
+		});
+
+		// Update lockfile
+		this.info(`Updating lockfile`);
+		await execa(`pnpm`, [`install`], {
+			cwd: releaseGroup.directory,
+		});
+
+		// Remove override after install
+		this.info(`Restoring package.json to original state`);
+		await execa(`git`, [`restore`, `--source=HEAD`, `package.json`], {
+			cwd: releaseGroup.directory,
+		});
+
+		// Install again to remove the override from the lockfile
+		this.info(`Updating lockfile to remove override`);
+		await execa(`pnpm`, [`install`], {
+			cwd: releaseGroup.directory,
+		});
```

Comment:
One important caveat here is that if the specified override version does not match the range declared by whichever package in the dependency graph that has the direct dependency on the package being updated, I expect this will probably bring it back in line with the range requested by the package with the dependency, so the version we specified in the command will not be the one in use at the end.

Couldn't come up with a good quick idea on how to check that (would need to look at the declared ranges in the lockfile and see if the one specified when invoking the command matches all of them?) but if anyone has suggestions, happy to try them out. Otherwise, this would rely a bit on code review to make sure that the final versions match the intended one, and even if that slips through review, it feels like the risk of that introducing a new problem is low? We'd just be updating to the latest version accepted by the package with the dependency, so might not address CVEs if that was the intention, but otherwise should be ok.